python/composio/tools/local/mathematical/actions/calculator.py

```diff
@@ -19,12 +21,54 @@ class CalculatorResponse(BaseModel):
 
 class Calculator(LocalAction[CalculatorRequest, CalculatorResponse]):
     """
-    Useful to perform any mathematical calculations, like sum, minus, multiplication, division, etc.
+    Performs mathematical calculations such as addition, subtraction, multiplication, and division.
     """
 
     _tags = ["calculator"]
 
+    # Define supported operators
+    operators = {
+        ast.Add: operator.add,
+        ast.Sub: operator.sub,
+        ast.Mult: operator.mul,
+        ast.Div: operator.truediv,
+        ast.Pow: operator.pow,
+        ast.USub: operator.neg,
+        ast.UAdd: operator.pos,
+    }
+
     def execute(self, request: CalculatorRequest, metadata: Dict) -> CalculatorResponse:
```

Comment:
Great job on addressing the security vulnerability by replacing `eval()` with a safer `ast`-based implementation. This significantly reduces the risk of code injection attacks. The new code structure using `ast` is more robust and easier to maintain.